### PR TITLE
First stab at end-to-end support for product id override on source

### PR DESCRIPTION
**Pull request by sturufous**
Due to a glitch on my end, the commit message is wrong, but this shouldn't affect the content. 

I rejigged the insert statements for the daily print papers in 01-Source.sql to avoid having to edit the other 246 entries.

### CODE REVIEW

File: app/editor/src/features/admin/sources/SourceDetails.tsx [hunk 1]
```diff
@@ -64,6 +65,12 @@ export const SourceDetails: React.FC<ISourceDetailsProps> = () => {
           tooltip="The user that manages this content"
           options={users}
         />
+        <FormikSelect
```
Comment (Fosol) [inline app/editor/src/features/admin/sources/SourceDetails.tsx:68]: This currently doesn't allow them to select None.

File: libs/net/dal/Configuration/SourceConfiguration.cs [hunk 2]
```diff
@@ -15,6 +15,7 @@ public override void Configure(EntityTypeBuilder<Source> builder)
         builder.Property(m => m.ShortName).IsRequired().HasMaxLength(100).HasDefaultValueSql("''");
         builder.Property(m => m.Description).IsRequired().HasMaxLength(2000).HasDefaultValueSql("''");
         builder.Property(m => m.OwnerId);
+        builder.Property(m => m.ProductId);
```
Comment (Fosol) [inline libs/net/dal/Configuration/SourceConfiguration.cs:18]: You should define the relationship as this is a foreign key.  See the examples below.

File: services/net/content/ContentManager.cs [hunk 3]
```diff
@@ -229,7 +229,8 @@ private async Task HandleMessageAsync(ConsumeResult<string, SourceContent> resul
                 // TODO: Failures after receiving the message from Kafka will result in missing content.  Need to handle this scenario.
                 // TODO: Handle e-tag.
                 var source = await _api.GetSourceForCodeAsync(model.Source);
-                if (model.ProductId == 0)
+                var productId = source.ProductId ?? model.ProductId;
```
Comment (Fosol) [inline services/net/content/ContentManager.cs:232]: Do not put this here.  The ingest that found this content should be making this decision on what the valid `ProductId` should be.

**Review by Fosol — CHANGES_REQUESTED** [review]

File: services/net/content/ContentManager.cs (hunk 3 above)
Comment (sturufous) [reply services/net/content/ContentManager.cs:232]: Fixed.

File: app/editor/src/features/admin/sources/SourceDetails.tsx (hunk 1 above)
Comment (sturufous) [reply app/editor/src/features/admin/sources/SourceDetails.tsx:68]: I followed the example of the other two dropdowns on the page. If the value of the field is null it displays "Select...". Once it's set you can change the value but can't set it back to null. What's the best way to set it to None in React?

File: app/editor/src/features/admin/sources/SourceDetails.tsx [hunk 4]
```diff
@@ -19,6 +19,7 @@ export const SourceDetails: React.FC<ISourceDetailsProps> = () => {
 
   const users = getUserOptions(lookups.users);
   const licenses = getSortableOptions(lookups.licenses);
+  const products = getSortableOptions(lookups.products);
```
Comment (Fosol) [inline app/editor/src/features/admin/sources/SourceDetails.tsx:22]: There is an override argument that allows you to prepend options.  You can prepend an option that handles `undefined` value.

**Review by Fosol — CHANGES_REQUESTED** [review]

File: app/editor/src/features/admin/sources/SourceDetails.tsx (hunk 1 above)
Comment (Fosol) [reply app/editor/src/features/admin/sources/SourceDetails.tsx:68]: I've left another comment about how to prepend an option to handle `undefined`

File: libs/net/dal/Configuration/SourceConfiguration.cs [hunk 5]
```diff
@@ -15,11 +15,13 @@ public override void Configure(EntityTypeBuilder<Source> builder)
...
         builder.Property(m => m.LicenseId).IsRequired();
         builder.Property(m => m.AutoTranscribe).IsRequired();
         builder.Property(m => m.DisableTranscribe).IsRequired();
 
         builder.HasOne(m => m.Owner).WithMany().HasForeignKey(m => m.OwnerId).OnDelete(DeleteBehavior.Restrict);
+        builder.HasOne(m => m.Category).WithMany().HasForeignKey(m => m.ProductId).OnDelete(DeleteBehavior.Restrict);
```
Comment (Fosol) [inline libs/net/dal/Configuration/SourceConfiguration.cs:24]: It's not a `Category`

File: libs/net/entities/Source.cs [hunk 6]
```diff
@@ -45,6 +45,17 @@ public class Source : BaseType<int>
...
+    /// <summary>
+    /// get/set - Foreign key to product.
+    /// </summary>
+    [Column("product_id")]
+    public int? ProductId { get; set; }
+
+    /// <summary>
+    /// get/set - The product associated with this data source..
+    /// </summary>
+    public Product? Category { get; set; }
```
Comment (Fosol) [inline libs/net/entities/Source.cs:57]: It's not a `Category`

**Review by Fosol — CHANGES_REQUESTED** [review]

**Comment by Fosol** [comment]
I believe once you rebase some of the comments I've added will already be resolved as long as you merge correctly.

File: libs/net/entities/Source.cs (hunk 6 above)
Comment (sturufous) [reply libs/net/entities/Source.cs:57]: I realized this wasn't an ideal name, but I couldn't call it Product as the class is Product. What name do you suggest?

File: libs/net/dal/Migrations/20221028170850_Initial.cs [hunk 7]
```diff
@@ -465,6 +465,12 @@ protected override void Up(MigrationBuilder migrationBuilder)
                         principalTable: "user",
                         principalColumn: "id",
                         onDelete: ReferentialAction.Restrict);
+                    table.ForeignKey(
```
Comment (Fosol) [inline libs/net/dal/Migrations/20221028170850_Initial.cs:468]: Remove your changes from commit.

File: libs/net/entities/Source.cs [hunk 8]
```diff
@@ -45,6 +45,12 @@ public class Source : BaseType<int>
     /// </summary>
     public virtual User? Owner { get; set; }
 
+    /// <summary>
+    /// get/set - Foreign key to product.
+    /// </summary>
+    [Column("product_id")]
```
Comment (Fosol) [inline libs/net/entities/Source.cs:51]: Remove your changes from commit.

**Review by Fosol — CHANGES_REQUESTED** [review]

File: libs/net/dal/Configuration/SourceConfiguration.cs (hunk 5 above)
Comment (Fosol) [reply libs/net/dal/Configuration/SourceConfiguration.cs:24]: Remove this change from your commit.

**Review by Fosol — CHANGES_REQUESTED** [review]

**Pull request closed without merging**